pkg/api/http_server.go

```diff
@@ -97,7 +98,9 @@ func (hs *HTTPServer) Run(ctx context.Context) error {
 
 	hs.log.Info("HTTP Server Listen", "address", listener.Addr().String(), "protocol", setting.Protocol, "subUrl", setting.AppSubUrl, "socket", setting.SocketPath)
 
-	hs.httpSrv = &http.Server{Addr: listenAddr, Handler: hs.macaron}
+	protect := csrf.Protect([]byte("1534a988296abc5e987s2700709gffca"))
```

Comment:
Should the `[]byte("32-byte-long-auth-key")` be unique for each deployment?  Does it make sense that every grafana deploy would have the same auth-key?

The docs for csrc say
```
    // PS: Don't forget to pass csrf.Secure(false) if you're developing locally
    // over plain HTTP (just don't leave it on in production).
```
how do we handle that?  perhaps we should trigger the csrf only when in "production" mode?

FYI -- i don't know anything about CSRF and am just reading the docs ;)

Reply:
Thanks for looking at this PR.

For the csrf.Secure(false) issue I'm hoping you guys can guide me.  I just started in the Grafana codebase, and I imagine there is some config/env var/flag we can key off to only do it in production or set csrf.Secure(false) in dev. It doesn't actually matter since we're not reading from the cookie on the client side. We're reading from the embedded value in the page js context.

For the 32-byte authKey arg value in csrf.Protect() I'm not sure it matters. I'll see if I can dig up some info on this. If it turns out that you're right then I have to change it in Sourcegraph too (my day job):-).

There's two other things we need to address before this is ready to land: 

- we need to make sure this doesn't disrupt API access.
- we need to solve the case when the value expires and needs to be renewed within a session. Right now I embed the value in the page. It's not clear to me how we should renew it if needed and avoid a page refresh which is ugly.

Reply:
For production/dev, you can use `if setting.Env != setting.PROD {`  
https://github.com/grafana/grafana/blob/master/pkg/models/context.go#L28